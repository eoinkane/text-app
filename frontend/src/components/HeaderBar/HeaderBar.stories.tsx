import React from "react";
import HeaderBarComponent from "./index";
import { linkTo } from "@storybook/addon-links";
import StoryRouter from "storybook-react-router";

import {
  LoadingContextProvider,
  useLoading,
} from "./../../contexts/LoadingContext";

export default {
  title: "Header Bar",
  component: HeaderBarComponent,
  decorators: [
    StoryRouter({
      "/HeaderBar": linkTo("Header Bar", "HeaderBar"),
    }),
  ],
};

export const Default = () => (
  <LoadingContextProvider>
    <HeaderBarComponent />
  </LoadingContextProvider>
);

const LoadingSpinnerOnHelper = () => {
  const { loadingActions } = useLoading("LoadingSpinner");
  loadingActions.ON();
  return <HeaderBarComponent />;
};

export const LoadingSpinnerOn = () => (
  <LoadingContextProvider>
    <LoadingSpinnerOnHelper />
  </LoadingContextProvider>
);
