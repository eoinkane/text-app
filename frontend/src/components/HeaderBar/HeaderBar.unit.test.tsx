import React from "react";
import { shallow } from "enzyme";
import HeaderBar from "./index";

import Typography from "@material-ui/core/Typography";
import Button from "@material-ui/core/Button";
import { MemoryRouter, Link } from "react-router-dom";

describe("HeaderBar", () => {
  it("Renders", () => {
    const headerBar = shallow(
      <MemoryRouter>
        <HeaderBar />
      </MemoryRouter>
    );
    expect(headerBar).toMatchSnapshot();
  });

  it("Renders with correct text", () => {
    const headerBar = shallow(<HeaderBar />);

    expect(headerBar).toMatchSnapshot();
    expect(headerBar.find(Typography).text()).toEqual("Messages");
    expect(headerBar.find(Button).text()).toEqual("Login");
  });

  it('Goes to "/" when home button pressed', () => {
    const headerBar = shallow(
      <MemoryRouter initialEntries={["/test_url"]}>
        <HeaderBar />
      </MemoryRouter>
    );

    headerBar.find(HeaderBar).dive().find(Link).simulate("click");

    expect(location.pathname).toBe("/");
  });
});
