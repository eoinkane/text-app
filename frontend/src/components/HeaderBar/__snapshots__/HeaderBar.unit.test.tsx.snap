// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar Renders 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "pathname": "/",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <ButtonAppBar />
</Router>
`;

exports[`HeaderBar Renders with correct text 1`] = `
<div
  className="makeStyles-root-1"
>
  <WithStyles(ForwardRef(AppBar))
    position="static"
  >
    <WithStyles(ForwardRef(Toolbar))>
      <WithStyles(ForwardRef(IconButton))
        aria-label="menu"
        className="makeStyles-menuButton-2"
        color="inherit"
        component={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "displayName": "Link",
            "propTypes": Object {
              "innerRef": [Function],
              "onClick": [Function],
              "replace": [Function],
              "target": [Function],
              "to": [Function],
            },
            "render": [Function],
          }
        }
        edge="start"
        to="/"
      >
        <MenuIcon />
      </WithStyles(ForwardRef(IconButton))>
      <WithStyles(ForwardRef(Typography))
        className="makeStyles-title-3"
        variant="h6"
      >
        Messages
      </WithStyles(ForwardRef(Typography))>
      <WithStyles(ForwardRef(Button))
        color="inherit"
      >
        Login
      </WithStyles(ForwardRef(Button))>
    </WithStyles(ForwardRef(Toolbar))>
  </WithStyles(ForwardRef(AppBar))>
</div>
`;
